### ISSUE TITLE
nomad registered service - script check wrongly registered as ttl check

### ISSUE THREAD

**Issue by eddiewtchoi** [hashicorp/consul#7718]
When filing a bug, please include the following headings if possible. Any example text in this template can be deleted.

#### Overview of the Issue

nomad registered service - script check wrongly registered as ttl check
service registered using nomad service, script check registered with raw_exec driver
nomad version 0.10.3
consul version 1.7.2 - enable_script_checks: true
*UI, not showing script output on ttl checks

#### Reproduction Steps

Steps to reproduce this issue, eg:

1.create nomad/consul cluster of 1 in dev mode
2.create dummie job myservice with script check 
3.curl -ks http://localhost:8500/v1/health/checks/myservice
type: ttl
### Operating system and Environment details
redhat linux 7
### Log Fragments

Include appropriate Client or Server log fragments. If the log is longer than a few dozen lines, please include the URL to the [gist](https://gist.github.com/) of the log instead of posting it in the issue. Use `-log-level=TRACE` on the client and server to capture the maximum log detail.

**Comment by johncowen** [comment]
Hi @eddiewtchoi 

I don't think I can help with what I understand is the main issue here (I'm sure someone on the backend team will pitch in and help here).

But for the UI not showing the script output on TTL checks, as far as I'm aware TTL checks don't have any output, so we explicitly turn off the Output field in the UI so it doesn't just show you the header with an empty value.

https://github.com/hashicorp/consul/blob/137a2c32c6e7ca6ca475f50be8a8c97532fa06ad/ui-v2/app/templates/components/healthcheck-list.hbs#L35

I hope at least that explains why you don't see the script output for TTL checks in the UI.

**Comment by eddiewtchoi** [comment]
more info on the case, i think is sensible for TTL check to not display output on UI, i think the problem is due to incorrect type registered, here are the debug job file / debug script / consul api output

job "consul-check-debug" {
    datacenters = ["dc1"]
    type = "service"
    group "consul-check-debug-group" {
        count = 1
        task "consul-check-debug-task" {
          driver = "raw_exec"
          logs {
            max_files     = 1
            max_file_size = 10
          }
          config {
            command = "tail"
            args    = [
                    "-f",
                    "/dev/null"
                     ]
          }
          service {
            name = "consul-check-debug"
            check {
              interval = "10s"
              timeout  = "2s"
              tls_skip_verify = true
              type     = "script"
              name     = "consul-check-debug"
              command  = "/var/tmp/consul-check-debug.sh"
            }
          }
        }
    }
}
 
# cat "/var/tmp/consul-check-debug.sh"
#/bin/bash
echo rc=1
exit 1
 
# curl -ks  https://localhost:8500/v1/health/checks/consul-check-debug?pretty
[
    {
        "Node": "masked-node-name",
        "CheckID": "_nomad-check-c694273f5dfb3b2c2f1842fa3204740512f83b0a",
        "Name": "consul-check-debug",
        "Status": "warning",
        "Notes": "",
        "Output": "rc=1\n",
        "ServiceID": "_nomad-task-606d149d-f93b-c0ef-cacb-2a506d0fb175-consul-check-debug-task-consul-check-debug-",
        "ServiceName": "consul-check-debug",
        "ServiceTags": [],
        "Type": "ttl",
        "Definition": {},
        "CreateIndex": 20526992,
        "ModifyIndex": 20527217
    }
]

**Comment by jorgemarey** [comment]
Hi, just some info about this (I'm not from the hashicorp team). When you define a script check in a nomad service stanza the nomad client will register is as ttl check and execute the script. As far as I'm aware this was always the case. The problem now is that consul from the 1.7.0 is not showing any output for TTL checks so that information is lost.

**Comment by eddiewtchoi** [comment]
> Hi, just some info about this (I'm not from the hashicorp team). When you define a script check in a nomad service stanza the nomad client will register is as ttl check and execute the script. As far as I'm aware this was always the case. The problem now is that consul from the 1.7.0 is not showing any output for TTL checks so that information is lost.

Yes, from a coding perspective, I think this is more of a nomad issue, which check should properly run as script type.

**Comment by jorgemarey** [comment]
I don't know if that should be the case... That would require operators to configure `enable_script_checks` in the consul configuration. I think that nomad should keep working the way it's. Maybe consul could show the TTL output in the UI (if there's output) again? If we use the API to get the healthchecks, the output information is there.

**Comment by eddiewtchoi** [comment]
> I don't know if that should be the case... That would require operators to configure `enable_script_checks` in the consul configuration. I think that nomad should keep working the way it's. Maybe consul could show the TTL output in the UI (if there's output) again? If we use the API to get the healthchecks, the output information is there.

The current way to register as ttl check, it seems like a quick workaround/hack around the enable_script_check. In my case, I fall back my consul implementation to 1.6.5 as I don’t need the new feature on 1.7.x
